src/spikeinterface/core/waveform_tools.py

```diff
@@ -397,6 +398,280 @@ def _waveform_extractor_chunk(segment_index, start_frame, end_frame, worker_ctx)
                     wfs[pos, :, :] = wf[:, sparsity_mask[unit_ind]]
 
 
+def extract_waveforms_to_single_buffer(
+    recording,
+    spikes,
```

Comment:
As we lack type for this can we name this variable spike_vector. Will be easier to ctrl+f in the library if someone wants to see the struct without having to read the code.

Reply:
`spikes` has been for this variable in many places since a while now. I would not chand this naming in this PR.
But wa can discuss in another one no ?

Reply:
That's fine. I find that spikes by itself is not specific enough for someone that is not familar with the library (is this a list? a dict? what kind of spikes?). I think that if you came up with that data structure you should leverage on it. Plus, if we ever want to refactor, it will be easier to locate the use cases of this structure.